### PR TITLE
feat: matomo: support premise version

### PR DESCRIPTION
**Pull request by anantjain45823**
## PR Description

Resolves INT-777
Through this we will start supporting premise version for matomo

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by anantjain45823** [comment]
used `no-verify` flag just to keep my commits on remote as I need to reclone my repo

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js [hunk 1]
```diff
@@ -172,14 +174,14 @@ class Matomo {
     }
 
     // Mapping Ecommerce Events
-    if (ecommerceMapping.has(trimmedEvent)) {
-      ecommerceEventsMapping(ecommerceMapping.get(trimmedEvent), message);
+    if (ecommerceMapping?.[trimmedEvent]) {
```
Comment (anantjain45823) [inline packages/analytics-js-integrations/src/integrations/Matomo/browser.js:177]: @MoumitaM @saikumarrs optional chaining would work fine here or shall I use `isDefinedAndNotNull`

**Comment by sanpj2292** [comment]
@anantjain45823 Can we add a test-case where in `matomoVersion` is `undefined` or `null` ?

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-js-integrations/src/integrations/Matomo/browser.js:177]: Yes this should work. If possible test it once.

Comment (anantjain45823) [reply packages/analytics-js-integrations/src/integrations/Matomo/browser.js:177]: Ok cool thanks! Will add a test case for this

Comment (MoumitaM) [inline packages/analytics-js-integrations/src/integrations/Matomo/browser.js:177]:
```suggestion
    const ecommerceEvent = ecommerceMapping[trimmedEvent];
    if (ecommerceEvent) {
```

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js [hunk 2]
```diff
@@ -172,14 +174,14 @@ class Matomo {
...
+      ecommerceEventsMapping(ecommerceMapping[trimmedEvent], message);
```
Comment (MoumitaM) [inline packages/analytics-js-integrations/src/integrations/Matomo/browser.js:178]:
```suggestion
      ecommerceEventsMapping(ecommerceEvent, message);
```

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js (hunk 1 above)
Comment (anantjain45823) [reply packages/analytics-js-integrations/src/integrations/Matomo/browser.js:177]: Done

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js (hunk 2 above)
Comment (anantjain45823) [reply packages/analytics-js-integrations/src/integrations/Matomo/browser.js]: Done

**Review by MoumitaM — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/Matomo/browser.js (hunk 1 above)
Comment (anantjain45823) [reply packages/analytics-js-integrations/src/integrations/Matomo/browser.js]: done

**Review by sanpj2292 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]
